packages/assets-controllers/src/NftController.ts

```diff
@@ -1497,21 +1510,27 @@ export class NftController extends BaseControllerV1<NftConfig, NftState> {
    * Refetches NFT metadata and updates the state
    *
    * @param options - Options for refetching NFT metadata
-   * @param options.nfts - Array of nfts
-   * @param options.networkClientId - The networkClientId that can be used to identify the network client to use for this request.
+   * @param options.nfts - nfts to update metadata for.
    * @param options.userAddress - The current user address
+   * @param options.networkClientId - The networkClientId that can be used to identify the network client to use for this request.
    */
   async updateNftMetadata({
     nfts,
-    networkClientId,
     userAddress = this.config.selectedAddress,
+    networkClientId,
   }: {
     nfts: Nft[];
-    networkClientId?: NetworkClientId;
     userAddress?: string;
+    networkClientId?: NetworkClientId;
   }) {
     const chainId = this.getCorrectChainId({ networkClientId });
-    const nftsWithChecksumAdr = nfts.map((nft) => {
+    // We want to update only nfts missing name/image or description
+    const nftsToUpdate = nfts.filter(
+      (singleNft) =>
+        !singleNft.name && !singleNft.description && !singleNft.image,
```

Comment:
I'm wondering if it would be better for the caller to perform this filter.  This would allow `updateNftMetadata`  to support NFTs that already have these fields. Since NFT metadata can change whenever, a caller may want to refresh for reasons other than the settings change.

Reply:
Totally! that makes sense to me, adding back the nfts argument was a step toward supporting more generic behaviors(updating nfts regardless whether they have name/image/description), i can have onPreferencesStateChange do the filter 🙏